### ISSUE TITLE
Bees with solid pollination blocks can become stuck due to their pathing

### ISSUE THREAD

**Issue by firebolt637** [Team-Resourceful/ResourcefulBees#73]
**Describe the bug**
Any bees that require a solid block to pollinate (e.g iron bee, diamond bee and so on) can become stuck due to how the bee's path. 

**To Reproduce**
Steps to reproduce the behaviour:
1. Place a bunch of bees that require a solid block for pollination in a controlled environment.
2. Place pollination blocks near the bees ensuring the block is raised one from the ground.
3. Wait until you see the bees fly over to the block and you will find that sometimes they will get stuck on the side of the block

**Expected behaviour**
So whats I belives to be happening here is when a been selects a block to "pollination" it will path over to the block and enter into the blocks same XYZ. This isn't an issue normally as "pollination" blocks are normally flowers meaning they can fly right through them as they have no hitbox but with "pollination" that are solid this does pose an issue. If the bee does not approach the block from the top it will try end fly through the side of the block and become stuck just pushing up against the block until something bumps the bee upwards where it can start to pollination.

To summarize the issue basically bees expect to be able to fly through their pollination block and when their pollination has a collider, they become stuck on the sides of the block every now and again and stay that way until the block is removed or bumped upwards.
 
**please complete the following information:**
 - Forge Version [e.g. '1.16.4 - 35.0.18']
 - Mod Version [e.g. '0.5.5b']

**Comment by Epic428** [comment]
Hey sorry for the late reply on this. It’s a pathfinding issue with Minecraft. It occasionally happens with vanilla bees as well. Imo Minecraft pathfinding is garbage and janky. I’m not 100% on the best way to solve it. Alternatively however if you put the blocks as the floor of your apiary they won’t be able to get stuck and will still pollinate.

**Comment by mkaito** [comment]
Putting blocks *in* the floor does indeed solve the problem. However, not all relevant blocks seem to be supported. I just came across Soul Sand (for nether warts) that wasn't recognized as a valid apiary block. There might be other relevant blocks. Granted, that's probably a custom bee problem.

I've also found it useful to put blocks far away from the apiary, as the bee path seems to dip down right after they exit the apiary, which leads to them getting easily stuck on blocks nearby.
